File: app/Http/Controllers/Helper/Pagination.js
'use strict'

class Pagination {
  constructor (request) {
    let page = Number(request.input('page', 1))
    let perPage = Number(request.input('per_page', 10))

    if (isNaN(page) || page < 1) {
      page = 1
    }

    if (isNaN(perPage) || perPage < 1) {
      perPage = 10
    }

    this.page = page
    this.perPage = perPage
    this.total = 0
    this.data = null
  }

  toJSON () {
    return {
      total: Number(this.total),
      perPage: this.perPage,
      currentPage: this.page,
      lastPage: Math.ceil(this.total / this.perPage),
      data: this.data
    }
  }
}

module.exports = Pagination

File: Sechzehn/app/src/main/java/de/tu_darmstadt/informatik/tk/ip/bravo/sechzehn/data/Pagination.java
package de.tu_darmstadt.informatik.tk.ip.bravo.sechzehn.data;

import android.support.annotation.NonNull;

import java.util.LinkedList;
import java.util.List;

/**
 * Provides a generic Wrapper for paginated responses
 * Created by JohannesTP on 19.07.2017.
 */

public class Pagination<T> {
    /**
     * The total number of elements
     */
    public int total;
    /**
     * The number of the elements per page
     */
    public int perPage;
    /**
     * The number of the current page
     */
    public int currentPage;
    /**
     * The number of the last page
     */
    public int lastPage;
    /**
     * Data of this Page
     */
    @NonNull
    public List<T> data = new LinkedList<>();
}
